pkg/registry/storage/volumeattachment/strategy.go

```diff
@@ -98,15 +106,23 @@ func (volumeAttachmentStrategy) PrepareForUpdate(ctx context.Context, obj, old r
 	if requestInfo, found := genericapirequest.RequestInfoFrom(ctx); found {
 		groupVersion = schema.GroupVersion{Group: requestInfo.APIGroup, Version: requestInfo.APIVersion}
 	}
+
+	newVolumeAttachment := obj.(*storage.VolumeAttachment)
+	oldVolumeAttachment := old.(*storage.VolumeAttachment)
+
 	switch groupVersion {
 	case storageapiv1beta1.SchemeGroupVersion:
 		// allow modification of Status via main resource for v1beta1
 	default:
-		newVolumeAttachment := obj.(*storage.VolumeAttachment)
-		oldVolumeAttachment := old.(*storage.VolumeAttachment)
 		newVolumeAttachment.Status = oldVolumeAttachment.Status
 		// No need to increment Generation because we don't allow updates to spec
 	}
+
+	// disable InlineVolumeSource
+	if !utilfeature.DefaultFeatureGate.Enabled(features.CSIInlineVolume) {
```

Comment:
to handle downgrade cases, we allow the field to continue being set if it was already set.  See [example](https://github.com/kubernetes/kubernetes/blob/720f84b6944ac1ec8db3b452b141d3778a5870bc/pkg/apis/storage/util/util.go#L26)

Reply:
See response in newer PR here - https://github.com/kubernetes/kubernetes/pull/74086#issuecomment-464188838